app/src/main/java/org/mifos/mobile/ui/fragments/RegistrationFragment.kt

```diff
@@ -41,195 +43,207 @@ class RegistrationFragment : BaseFragment(), RegistrationView {
         _binding = FragmentRegistrationBinding.inflate(inflater, container, false)
         val rootView = binding.root
         (activity as BaseActivity?)?.activityComponent?.inject(this)
-        presenter?.attachView(this)
-        binding.progressBar.visibility = View.GONE
-        binding.passwordStrength.visibility = View.GONE
-        binding.etPassword.addTextChangedListener(object : TextWatcher {
-            override fun beforeTextChanged(charSequence: CharSequence, i: Int, i1: Int, i2: Int) {}
-            override fun onTextChanged(charSequence: CharSequence, i: Int, i1: Int, i2: Int) {
-                if (charSequence.isEmpty()) {
-                    binding.progressBar.visibility = View.GONE
-                    binding.passwordStrength.visibility = View.GONE
-                } else {
-                    binding.progressBar.visibility = View.VISIBLE
-                    binding.passwordStrength.visibility = View.VISIBLE
-                    updatePasswordStrengthView(charSequence.toString())
+        viewModel = ViewModelProvider(this, viewModelFactory)[RegistrationViewModel::class.java]
+        with(binding) {
+            progressBar.visibility = View.GONE
+            passwordStrength.visibility = View.GONE
```

Comment:
The initial value of View.GONE can be set in the XML file as well, no need to add here, remove it